### DISCUSSION TITLE
Bitwarden_rs crashes

### DISCUSSION THREAD

**Discussion by 1dilemma2** [dani-garcia/vaultwarden#1616] [Q&A]
### Subject of the issue
bitwarden_rs crashes in some occasions

### Deployment environment

Synology DS 1513+, Chrome plugin
    =========================================================================================
    Preferably, use the `Generate Support String` button on the admin page's Diagnostics tab.
    That will auto-generate most of the info requested in this section.
    =========================================================================================
-->

* bitwarden_rs version: 1.20.0 (can't access Diagnostics tab. Error message: page not found)

* Install method: Docker image

* Clients used:web vault
* 
* Reverse proxy and version: up to date DSM
* 
* MySQL/MariaDB or PostgreSQL version: up to date package

* Other relevant details:

### Steps to reproduce
Wanted to create a folder in my tresor. Somtimes it works. most time not. Pressed "create folder", entered folder name, pressed OK. Worked a long time.

### Expected behaviour
Folder should be created

### Actual behaviour
Error message most of the time I tried. Container has to be restarted to work again.

### Troubleshooting data
Error message: 
![BW](https://user-images.githubusercontent.com/82753077/115140197-5504fb00-a036-11eb-9012-f6f4dc4bdb77.jpg)

Today I managed to create 2 folders. After that I wanted to delete both. But it was only possible to delete one. After pressing "Delete" for the second folder the same error message appears after a while and the container needs to be restarted again.

**Comment by BlackDex** [comment]
It looks like your reverse proxy is blocking PUT or DELETE requests (Since it returns 405).
Bitwarden_RS needs GET, POST, PUT, DELETE, without those HTTP Methods the application will not work.

**Comment by 1dilemma2** [comment]
If it blocks it...why does it work sometimes? Why does it bring bitwarden_rs to crash?

And much more interesting...How can I allow GET, POSR, PUT, DELETE? Because I didn't block anything.

**Comment by BlackDex** [comment]
What is the reason you think Bitwarden_RS crashes? I do not see that in the logs and screenshots you provided.

I have no clue how the proxy software of a Synology works, so i can't really help with that, other then trying to search for it my self on the internet.
